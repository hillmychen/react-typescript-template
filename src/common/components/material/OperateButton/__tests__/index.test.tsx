import React from 'react'
import { render, cleanup, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { OperateButton } from '../index'

afterEach(cleanup)

it('disabled=false', () => {
  const { asFragment } = render(<OperateButton style={{ padding: 2 }} onClick={() => console.log('onClick')} />)
  expect(asFragment()).toMatchSnapshot()
})

it('disabled=true', () => {
  const { asFragment } = render(<OperateButton disabled />)
  expect(asFragment()).toMatchSnapshot()
})

it('OperateButton click loading', () => {
  const onClick = jest.fn().mockImplementation(() => Promise.resolve())
  const { asFragment, getByTestId } = render(<OperateButton onClick={onClick}>点击</OperateButton>)
  userEvent.click(getByTestId('OperateButton'))
  expect(onClick).toBeCalled()
  expect(asFragment()).toMatchSnapshot()
})

it('OperateButton click no loading', async () => {
  const onClick = jest.fn().mockImplementation(() => Promise.resolve())
  const { asFragment, getByTestId } = render(<OperateButton onClick={onClick}>点击</OperateButton>)
  await waitFor(() => userEvent.click(getByTestId('OperateButton')))
  expect(onClick).toBeCalled()
  expect(asFragment()).toMatchSnapshot()
})
