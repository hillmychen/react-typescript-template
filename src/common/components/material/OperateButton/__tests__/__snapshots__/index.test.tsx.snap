// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperateButton click loading 1`] = `
<DocumentFragment>
  <a
    class="operateButton"
    data-testid="OperateButton"
  >
    点击
    <div
      class="ant-spin ant-spin-sm ant-spin-spinning spin"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </a>
</DocumentFragment>
`;

exports[`OperateButton click no loading 1`] = `
<DocumentFragment>
  <a
    class="operateButton"
    data-testid="OperateButton"
  >
    点击
    <div
      class="ant-spin ant-spin-sm spin"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </a>
</DocumentFragment>
`;

exports[`disabled=false 1`] = `
<DocumentFragment>
  <a
    class="operateButton"
    data-testid="OperateButton"
    style="padding: 2px;"
  >
    <div
      class="ant-spin ant-spin-sm spin"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </a>
</DocumentFragment>
`;

exports[`disabled=true 1`] = `
<DocumentFragment>
  <a
    class="operateButton operateButtonDisabled"
    data-testid="OperateButton"
  >
    <div
      class="ant-spin ant-spin-sm spin"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </a>
</DocumentFragment>
`;
